hw/dv/verilator/cpp/scrambled_ecc32_mem_area.cc

```diff
@@ -0,0 +1,161 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "scrambled_ecc32_mem_area.h"
+#include "scramble_model.h"
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <sstream>
+
+#include "sv_scoped.h"
+
+// This is the maximum width of a nonce that's supported by the code in
+// prim_util_get_scramble_key_nonce.svh
+static const uint32_t kScrMaxNonceWidth = 320;
+static const uint32_t kScrMaxNonceWidthBytes = (kScrMaxNonceWidth + 7) / 8;
+
+// Functions to convert from integer address to/from a vector of bytes
+static std::vector<uint8_t> AddrIntToBytes(uint32_t addr) {
+  std::vector<uint8_t> addr_bytes(4);
+
+  addr_bytes[0] = addr & 0xff;
+  addr_bytes[1] = (addr >> 8) & 0xff;
+  addr_bytes[2] = (addr >> 16) & 0xff;
+  addr_bytes[3] = (addr >> 24) & 0xff;
+
+  return addr_bytes;
+}
+
+static uint32_t AddrBytesToInt(const std::vector<uint8_t> &addr) {
+  uint32_t addr_out = 0;
```

Comment:
I think this needs an assertion that `addr.size() == 4` (or maybe switch to using `std::array<uint8_t, 4>` to move the assertion to compile-time)

Reply:
Assertion is `addr.size() <= 4` as it can process address vectors of 1, 2 and 3 bytes also.